runtime/src/commonTest/kotlin/pbandk/AnyTest.kt

```diff
@@ -0,0 +1,68 @@
+package pbandk
+
+import pbandk.testpb.Foo
+import pbandk.testpb.TestAllTypesProto3
+import pbandk.wkt.Any
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertFalse
+import kotlin.test.assertTrue
+
+class AnyTest {
+    @Test
+    fun testAny_field() {
+        val message = TestAllTypesProto3(optionalAny = Any.pack(TestAllTypesProto3(optionalInt32 = 12345)))
+        val encoded = message.encodeToByteArray()
+        
+        assertEquals(message, TestAllTypesProto3.decodeFromByteArray(encoded))
+    }
+
+    @Test
+    fun testAny_topLevel() {
+        val message = Any.pack(TestAllTypesProto3(optionalBool = true))
+        val encoded = message.encodeToByteArray()
+
+        assertEquals(message, Any.decodeFromByteArray(encoded))
+    }
+    
+    @Test
+    fun test_isA_positive() {
+        val message = Any.pack(TestAllTypesProto3(optionalInt32 = 12345))
+
+        assertTrue(message.isA(TestAllTypesProto3))
+    }
+    
+    @Test
+    fun test_isA_negative() {
+        val message = Any.pack(Foo(`val` = "hi"))
+
+        assertFalse(message.isA(TestAllTypesProto3))
+    }
+    
+    @Test
+    fun test_isA_customTypeUrl() {
```

Comment:
We could also include a test for a typeUrlPrefix without the trailing "/".

Reply:
Good idea 👍🏼  Done.